RIOT-OS/RIOT
# cpu/stm32: Restructure stm32 clk kconfig model

### Contribution description

Currently nightlies are failing due to kconfig hash mismatches, ultimately due to clock tree configurations.
It is pretty hard to understand what is happening so this restructuring should be a first step to fixing the current issues (essentially not enough information is given to kconfig such as the HSE crystal speed, to make as good a choice for the clock settings are make).

This PR splits the Kconfig clock configurations for the stm32s from one file for all series, into many each series and any common combinations of series. This makes it more readable and matches what is done in make a bit closer.

As only the series that is applicable gets sourced, it also removes a lot of noise created by other series configurations being included.

### Testing procedure

Murdock with nightlies enabled should have the same errors and passes as the latest nightlies.

Locally, all stms can be tested for an app with the following few lines:

Get a variable to hold all stm32 based boards
```
export TEST_BOARDS=$(FEATURES_REQUIRED=cpu_stm32 make -C tests/shell info-boards-supported --no-print-directory)
```

Overwrite the nightly flag to run all kconfig tests and binary comparisons with make
```
export NIGHTLY=1
```

Run through each stm based board for a single test, doing the kconfig comparison
```
for board in $TEST_BOARDS; do echo $board && /bin/bash -c "source .murdock; JOBS=4 compile tests/shell ${board}:gnu" | grep mismatch; done;
```

it should take maybe 10 to 20 mins, still faster than waiting for the CI.

### Issues/PRs references
